### PR TITLE
Feature/reimplement tabs

### LINKED ISSUE #556: Re-implement tab control
Currently disabled as part of redesign.

Should set and manage tabs from home and map views.

Now have three tabs instead of two: have added 'learn' tab.

### PR DESCRIPTION
**Pull request by CloudNiner**
## Overview

This PR adds in a new tab control (replacing the unnecessary sidebar tab control) which controls in page state changes between the home/directions/explore tabs. It's a bit awkward, because those states are controlled in page, but the learn tab goes to a new Django view.

## Testing

I _think_ I covered all of the interactions, but I might have missed something since I'm still a bit green on this project. Click back and forth between the map/directions tab and make sure things work, at various screen sizes.

Things that do not work:
- If you click the 'Explore' tab on the home view, you'll get a broken view. This should be cleaned up in #593 

Supersedes #624 
Closes #556

### CODE REVIEW

File: src/app/scripts/cac/map/cac-map-control.js [hunk 1]
```diff
@@ -163,47 +148,39 @@ CAC.Map.Control = (function ($, Handlebars, cartodb, L, turf, _) {
...
         //overlays['Nearby Events'] = overlaysControl.nearbyEventsOverlay();
         //overlays['Nearby Events'].addTo(map);
     }
 
     function initializeLayerControl() {
-        layerControl = cartodb.L.control.layers(basemaps, overlays, {
-            position: 'bottomright',
-            collapsed: false
-        });
+        if (!layerControl) {
```
Comment (flibbertigibbet) [inline src/app/scripts/cac/map/cac-map-control.js]: Noting here that the changes for this section will conflict with #625 so we'll need to fix that up

File: src/app/scripts/cac/pages/cac-pages-home.js [hunk 2]
```diff
@@ -129,14 +130,24 @@ CAC.Pages.Home = (function ($, ModeOptions,  MapControl, Modal, Templates, UserP
...
+            }
+        });
+        $(options.selectors.homeLink).on('click', function (event) {
+            event.preventDefault();
+            event.stopPropagation();
+
+            tabControl.setTab(tabControl.TABS.HOME);
+        });
+
+        $(document).ready(directionsControl.setFromUserPreferences());
```
Comment (flibbertigibbet) [inline src/app/scripts/cac/pages/cac-pages-home.js]: Might want to leave a TODO in here about user preferences going away/needing refactor still

**Comment by flibbertigibbet** [comment]
Spun it up, LGTM :steam_locomotive:

**Comment by CloudNiner** [comment]
Great. Going to merge this, happy to clarify any issues with merge conflicts on your layer switcher control PR.

**Pull request merged**